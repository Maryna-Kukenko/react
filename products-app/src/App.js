import React from 'react';
import {Container} from "react-bootstrap";
import ProductsView from "./container/ProductsView";

function App() {
  return (
      <Container>
        <ProductsView />
      </Container>
  );
}

export default App;