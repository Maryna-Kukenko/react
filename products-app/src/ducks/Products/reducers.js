import * as Types from './actionTypes'

const store = {
  products: [],
  categories: []
};

const reducers = (state = [], action) => {
  switch (action.type) {
    case Types.GET_PRODUCTS:
      return [
        ...state,
        action.payload
        ];
    case Types.GET_CATEGORIES:
      return [
        ...state,
        action.payload
      ];
    default: {
      return state
    }
  }
};

export default reducers;