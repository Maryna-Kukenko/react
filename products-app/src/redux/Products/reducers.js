import * as Types from './actionTypes'

const store = {
  products: [],
  categories: []
};

const reducers = (state = store, action) => {
  switch (action.type) {
    case Types.GET_PRODUCTS:
      return {
        ...state,
        products: action.payload
      };
    case Types.GET_CATEGORIES:
      return {
        ...state,
        categories: action.payload
      };
    default: {
      return state
    }
  }
};

export default reducers;