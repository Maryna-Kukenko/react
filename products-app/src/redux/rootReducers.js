import { combineReducers} from 'redux';
import reducers from './Products/reducers';

export default combineReducers({
  reducers
})