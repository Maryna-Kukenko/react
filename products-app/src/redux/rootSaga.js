import * as Types from './Products/actionTypes'
import addData from './Products/getDataSaga'
import { takeEvery } from '@redux-saga/core/effects';

export default function* getData() {
  yield takeEvery(Types.GET_DATA, addData)
}