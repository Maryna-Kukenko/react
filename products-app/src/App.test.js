import React from 'react';
import ReactDOM from 'react-dom';
import { Router} from 'react-router';
import { Provider } from 'react-redux';
import {applyMiddleware, combineReducers, createStore} from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createBrowserHistory } from 'history';
import { composeWithDevTools } from 'redux-devtools-extension';
import  {shallow}  from 'enzyme';

import reducers  from './ducks/products';
import { App } from './App';

const history = createBrowserHistory();
const sagaMiddleware = createSagaMiddleware();
const rootReducers = combineReducers({reducers});

const store = createStore(
  rootReducers,
  composeWithDevTools(
    applyMiddleware(sagaMiddleware)
  )
);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}>
    <Router history={history}>
      <App />
    </Router>
  </Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Should render App component and match to snapshot', () => {
    const rendered = shallow(<Provider store={store}>
      <Router history={history}>
        <App/>
      </Router>
    </Provider>);
    expect(rendered).toMatchSnapshot();
  }
);