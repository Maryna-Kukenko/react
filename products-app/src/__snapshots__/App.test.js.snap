// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App component and match to snapshot 1`] = `ShallowWrapper {}`;
